socs/agents/scpi_psu/agent.py

```diff
@@ -80,8 +107,13 @@ def monitor_output(self, session, params=None):
                     }
 
                     for chan in params['channels']:
-                        data['data']["Voltage_{}".format(chan)] = self.psu.get_volt(chan)
-                        data['data']["Current_{}".format(chan)] = self.psu.get_curr(chan)
+                        if self.psu.get_output(chan):
+                            data['data']["Voltage_{}".format(chan)] = self.psu.get_volt(chan)
+                            data['data']["Current_{}".format(chan)] = self.psu.get_curr(chan)
+                        else:
+                            self.log.warn("Cannot measure output when output is disabled")
+                            self.monitor = False
+                            return False, "Cannot measure output when output is disabled"
```

Comment:
What's the response from the PSU when you query a disabled channel?

Reply:
at least with the 2280S, it sets an internal error flag and doesn't otherwise respond. So when you subsequently call `socket.recv()` to get the response it results in a timeout error -- less than an ideal way of handling that case, imo.

Reply:
Ah, alright, then we should add some error handling for that timeout error, else a query on a disabled channel will crash this process.

Reply:
As written, this will check to make sure the output is enabled before attempting a measurement in the `monitor_output()` process. I will add a similar check in the `get_voltage()` and `get_current()` functions. 

For the 2280S, anytime a command results in an error the next `socket.recv()` call will result in a timeout error. Ideally, the agent code should never send anything that would cause such an error (i.e. the agent should never send malformed commands or out-of-bounds parameter values, etc.). If a timeout error does occur, then it should indicate a networking issue rather than a bad command.

That said, I think the only time anyone should run into a timeout error caused by a bad command (e.g. a measurement command when the output is off, or setting the voltage on a channel that doesn't exist) should be if they are calling the driver functions directly. I also don't know if other SCPI power supplies respond the same way to bad commands.

Reply:
That makes sense. My concern is with the possibility of network issues. 

We don't handle network errors well in many of the agents, but are [trying to fix this](https://github.com/simonsobs/socs/issues/721). There is a simultaneous effort to make this agent robust to network errors too: https://github.com/simonsobs/socs/pull/726

See this discussion for more context: https://github.com/simonsobs/socs/discussions/538

Essentially we'd like to keep the agent online and have it reconnect to the device. In its current form the `monitor` process would crash if there's a Timeout. You'd then have to run `init` and then restart `monitor` after resolving the networking issue.